### PR TITLE
Improved Tests for Schema Size Limit (Issue #312)(Adding Test max encoded schema limit exceeded to pallet/schema)

### PR DESCRIPTION
**Pull request by HarishKarthickS**
Hi everyone,

This pull request addresses issue #312 and aims to improve the testing of the maximum schema size limit. These improvements ensure that the system correctly handles scenarios where the schema size might be exceeded.

My Changes:

I've added more comprehensive tests to the existing suite:
One test specifically checks the Schema::create function to verify it throws the MaxEncodedSchemaLimitExceeded error when data is too large.
Another test case focuses on an edge case, making sure the limit check works correctly for data sizes very close to the maximum allowed size.
I've used a mock storage system (MockStorage) to isolate the tests from external dependencies, making them more reliable.
I've included comments throughout the code to explain the purpose of each section (arrange, act, assert) for better understanding.
Benefits:

These additional tests will help us catch potential issues related to exceeding the schema size limit before they occur in production, leading to a more robust system.
Developers can be more confident in how the system handles large schemas.
Clear test cases will make it easier to maintain the code in the future.
Next Steps:

I'd appreciate it if you could review the code changes and test them.
If there are other functions that might be affected by the schema size limit, we could consider adding similar test cases for those as well.
Please double-check that the expected error type (MaxEncodedSchemaLimitExceeded) is accurate.
Thank you for your time and consideration. Please merge this pull request when you're satisfied with the changes.

Best,

Harish Karthick S

### CODE REVIEW

File: pallets/schema/src/tests.rs [hunk 1]
```diff
@@ -314,3 +314,48 @@ fn test_schema_lookup() {
 		}
 	});
 }
+
+#[cfg(test)]
```
Comment (smohan-dw) [inline pallets/schema/src/tests.rs]: It is better to extend the existing framework with new tests than create a mod. The #[cfg(test)] is activated with #[test] macro. Also, rather than passing a custom MAX_SCHEMA_SIZE, use the MaxEncodedSchemaLength value from mock.

Comment (HarishKarthickS) [reply pallets/schema/src/tests.rs]: Sir, I have made [Changes according to the review](https://github.com/dhiway/cord/pull/435/commits/3387095bd40868e0a41ae829290f22206c8822e4)

File: pallets/schema/src/tests.rs [hunk 2]
```diff
@@ -314,3 +314,30 @@ fn test_schema_lookup() {
...
 	});
 }
+
+#[test]
+fn test_max_encoded_schema_limit_exceeded_create() {
+	// Arrange
+	let mut mock_storage = MockStorage::new();
+
+	// Act
+	let large_schema_data = vec![0u8; <Test as frame_system::Config>::MaxEncodedSchemaLength + 1];
```
Comment (vatsa287) [inline pallets/schema/src/tests.rs:324]: @HarishKarthickS I don't think this can be done, as the input is compiled to be of type `BoundedVec` and not just `Vec`, where `MaxEncodedSchemaLength + 1` is impossible due to below,
`pub type InputSchemaOf<T> = BoundedVec<u8, <T as Config>::MaxEncodedSchemaLength>;
`
but it is possible for `MaxEncodedSchemaLength - 1` to take place with a `BoundedVec` since it will be below threshold input set limit.

**Comment by VedantKhairnar** [comment]
@HarishKarthickS 
Are you working on this PR?
